### PR TITLE
Optimize all-in-one image

### PR DESCRIPTION
**Pull request by mkumatag**
- Removed the Bazel
- Updated
  - ibmcloud cli
  - golang version
  - goversion
  - kubectl
  - base image to debian:bullseye
  - terraform
  - terraform modules - ibm, null
- Building TF in staging to reduce the image size
- Installing ansible from the debian distro
- Refactored to help multistage build
- Copying IBM Cloud terraform binary to `${TF_PLUGIN_PATH}/IBM-Cloud/ibm` as well to work with the latest provider block:
```hcl
terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
      version = "1.46.0-beta0"
    }
  }
}

provider "ibm" {
  # Configuration options
}
```

## Size reduced:

Before:
```shell
quay.io/powercloud/all-in-one   0.4        3bbafb96b501   6 months ago     6.04GB
```

After:
```shell
quay.io/powercloud/all-in-one   latest     af91351fd75d   15 minutes ago   3.11GB
```

### CODE REVIEW

File: images/all-in-one/Dockerfile [hunk 1]
```diff
@@ -89,16 +90,10 @@ RUN set -eux; \
...
-            echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | tee /etc/apt/sources.list.d/bazel.list; \
-            apt update && apt install -y bazel=$BAZEL_VERSION; \
             ;; \
         'ppc64el') \
             url="https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-ppc64le.tar.gz"; \
             ibm_cloud_url="https://download.clis.cloud.ibm.com/ibm-cloud-cli/${IBM_CLOUD_VERSION}/IBM_Cloud_CLI_${IBM_CLOUD_VERSION}_ppc64le.tar.gz"; \
-            wget -O /usr/local/bin/bazel "https://oplab9.parqtec.unicamp.br/pub/ppc64el/bazel/ubuntu_18.04/bazel_bin_ppc64le_$BAZEL_VERSION" --progress=dot:giga; \
-            chmod +x /usr/local/bin/bazel; \
             ;; \
         's390x') \
```
Comment (Rajalakshmi-Girish) [inline images/all-in-one/Dockerfile:98]: As mentioned [here](https://github.ibm.com/powercloud/container-dev/issues/1573#issuecomment-43603570), seems we may remove the s390x-related code

File: images/all-in-one/Dockerfile [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
-ARG IBM_CLOUD_VERSION=2.5.0
+ARG GOLANG_VERSION=1.18.6
+ARG IBM_CLOUD_VERSION=2.10.0
+ARG TERRAFORM_PROVIDER_IBM_VERSION=1.39.1
+ARG TERRAFORM_PROVIDER_NULL_VERSION=3.1.0
 FROM golang:$GOLANG_VERSION AS builder
 
-RUN go get rsc.io/goversion \
+RUN go install rsc.io/goversion@v1.2.0 \
     && GO111MODULE=on
```
Comment (Rajalakshmi-Girish) [inline images/all-in-one/Dockerfile:9]: As mentioned by golang toolchain team member [here](https://github.ibm.com/powercloud/container-dev/issues/1573#issuecomment-44085992), should we consider setting `GOPPC64=power9`?

**Review by Rajalakshmi-Girish — CHANGES_REQUESTED** [review]

File: images/all-in-one/Dockerfile (hunk 2 above)
Comment (mkumatag) [reply images/all-in-one/Dockerfile:9]: That won't make much difference here!

Note: I also need to remove that `&& GO111MODULE=on` which is not required.

File: images/all-in-one/Dockerfile (hunk 1 above)
Comment (mkumatag) [reply images/all-in-one/Dockerfile:98]: yes, lemme remove the whole block itself.

**Comment by mkumatag** [comment]
@Rajalakshmi-Girish fixed the review comments and added more information in the description, ptal

**Review by Rajalakshmi-Girish — APPROVED** [review]
/lgtm

**Pull request merged**